### PR TITLE
Material for DevCon SD2.1 Optimum-Intel

### PR DESCRIPTION
**Pull request by paularamo**

### CODE REVIEW

File: notebooks/236-stable-diffusion-v2/236-stable-diffusion-v2-optimum-demo.ipynb [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    "![image](https://github.com/openvinotoolkit/openvino_notebooks/assets/10940214/e166f225-1220-44aa-a987-84471e03947d)\n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "2ade39f1",
+   "metadata": {},
+   "source": [
+    "The base model used for this example is the \"[stabilityai/stable-diffusion-2-1-base](https://huggingface.co/stabilityai/stable-diffusion-2-1). This model was converted to OpenVINO format, for accelerated inference on CPU or Intel GPU with OpenVINO's integration into Optimum: optimum-intel. The model weights are stored with FP16 precision, which reduces the size of the model by half. You can find the model used in this notebook is \"[helenai/stabilityai-stable-diffusion-2-1-base-ov](https://huggingface.co/helenai/stabilityai-stable-diffusion-2-1-base-ov)\". Let's download the pre-converted model Srable Diffusion 2.1 [Intermediate Representation Format (IR)](https://docs.openvino.ai/2022.3/openvino_docs_MO_DG_IR_and_opsets.html)"
```
Comment (helena-intel) [inline notebooks/236-stable-diffusion-v2/236-stable-diffusion-v2-optimum-demo.ipynb:58]:
```suggestion
    "The base model used for this example is the \"[stabilityai/stable-diffusion-2-1-base](https://huggingface.co/stabilityai/stable-diffusion-2-1). This model was converted to OpenVINO format, for accelerated inference on CPU or Intel GPU with OpenVINO's integration into Optimum: optimum-intel. The model weights are stored with FP16 precision, which reduces the size of the model by half. You can find the model used in this notebook is \"[helenai/stabilityai-stable-diffusion-2-1-base-ov](https://huggingface.co/helenai/stabilityai-stable-diffusion-2-1-base-ov)\". Let's download the pre-converted model Stable Diffusion 2.1 [Intermediate Representation Format (IR)](https://docs.openvino.ai/2022.3/openvino_docs_MO_DG_IR_and_opsets.html)"
```

**Review by helena-intel — COMMENTED** [review]
Both notebooks worked great on my Linux instance with dGPU! I like how easy they make experimenting with different prompts and settings. And dGPU is so fast that that's actually fun. I have some small comments, none of them should block merging. 

Have you considered adding a link to your dGPU driver blog to the notebooks?

**Review by adrianboguszewski — COMMENTED** [review]
It (demo) doesn't work for me:
image
Maybe this is due to available hardware:
```
CPU: 12th Gen Intel(R) Core(TM) i7-12700H
GPU.0: Intel(R) Graphics [0x46a6] (iGPU)
GPU.1: NVIDIA GeForce RTX 3060 Laptop GPU (dGPU)
```
Also, I think more descriptions are needed. I cannot clearly state the difference between the two notebooks being merged in this PR. As a person, not being an author, I don't know what is the goal of any of them.

File: notebooks/236-stable-diffusion-v2/236-stable-diffusion-v2-optimum-demo-comparison.ipynb [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    "output_cpu_ov"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "d4444a06",
+   "metadata": {},
+   "source": [
+    "### Using FP16 precision model in dGPU with StableDiffusionPipeline\n"
```
[the remark this answers is not in the document]
Comment (paularamo) [reply notebooks/236-stable-diffusion-v2/236-stable-diffusion-v2-optimum-demo-comparison.ipynb:144]: Great add! Thanks Helena

**Comment by paularamo** [comment]
> It (demo) doesn't work for me: image Maybe this is due to available hardware:
> 
> ```
> CPU: 12th Gen Intel(R) Core(TM) i7-12700H
> GPU.0: Intel(R) Graphics [0x46a6] (iGPU)
> GPU.1: NVIDIA GeForce RTX 3060 Laptop GPU (dGPU)
> ```
> 
> Also, I think more descriptions are needed. I cannot clearly state the difference between the two notebooks being merged in this PR. As a person, not being an author, I don't know what is the goal of any of them.
> 
> Downloading all of these models is extremely slow (~20 minutes in my case).

Thanks Adrian. Could you please run it again?

**Comment by OpenVINO-dev-contest** [comment]
@paularamo Thanks for your update, I tried it on Flex server, and results look good.
I just wonder if we shall compare the performance in this demo? the units to measure the performance are not same for me.
- CPU
image

- OV_CPU
image

-OV_GPU
image

**Comment by paularamo** [comment]
```diff
- from diffusers import StableDiffusionPipeline
+ from optimum.intel.openvino import OVStableDiffusionPipeline

model_id = "stabilityai/stable-diffusion-2-1-base"
- pipe = StableDiffusionPipeline.from_pretrained(model_id)
+ pipe = OVStableDiffusionPipeline.from_pretrained(model_id, export=True, compile=False)
- pipe.save_pretrained("./stabilityai_cpu")
+ pipe.save_pretrained("./openvino_ir")
Pipe.compile()
prompt = "red car in snowy forest"
output_cpu = pipe(prompt, num_inference_steps=17).images[0]

```

**Pull request merged**